consensus/misc/rewards.go

```diff
@@ -62,3 +72,35 @@ func FindMinDenominations(reward *big.Int) map[uint8]uint8 {
 
 	return denominationCount
 }
+
+// calculateExchangeRate calculates the exchange rate for the block
+func CalculateExchangeRate(parent *types.WorkObject) (*big.Int, *big.Int, *big.Int, error) {
+	qiToQuai := parent.Header().QiToQuai() //Units of Qi
+	quaiToQi := parent.Header().QuaiToQi() //Units of Quai
+
+	//Calculate the new amounts of qi and quai being exchanged
+	for _, etx := range parent.ExtTransactions() {
+		if etx.Conversion() {
+			if etx.To().IsInQuaiLedgerScope() {
+				//Add the value of the transaction to the total amount of Qi being converted to Quai in Quai
+				qiToQuai = new(big.Int).Add(etx.Value(), qiToQuai)
+			} else if etx.To().IsInQiLedgerScope() {
+				//Add the value of the transaction to the total amount of Quai being converted to Qi in Qi
+				quaiToQi = new(big.Int).Add(etx.Value(), quaiToQi)
```

Comment:
you have used the same variable that you have initialized in the beginning of the function, and below you are adding the last exchange to the total qiToQuai and quaiToQi, is this correct?